### ISSUE TITLE
Lease heartbeat timeout, switch to Duration?

### ISSUE THREAD

**Issue by chbatey** [akka/akka#27082]
Lease is API may change until 2.6, should we change the heartbeat timeout to a Duration rather than a FiniteDuration? The safest value for this value is Duration.Inf which perhaps should be the default. WDYT @patriknw @jrudolph ?

**Comment by jrudolph** [comment]
The docs for that setting say

```
# if the node that acquired the leases crashes, how long should the lease be held before another owner can get it
```

Would disabling the timeout mean that nothing else can ever get a lease after a node has crashed without returning the leases?

**Comment by chbatey** [comment]
> Would disabling the timeout mean that nothing else can ever get a lease after a node has crashed without returning the leases?

Yes. E.g. an operator would need to come and clear the lease. It is inconvenient but the safest default. Then users explicitly opt automatic expiry and have to think about the timeout value

**Comment by patriknw** [comment]
If we really want to go with that default we should make it a mandatory property so that system will not be started without user thinking about it and putting in a specific value.

Current default is 2 minutes, which seems pretty safe for GC.
